src/consensus/aft/raft.h

```diff
@@ -1734,7 +1754,7 @@ namespace aft
       j["from_node_id"] = from;
       RAFT_TRACE_JSON_OUT(j);
 #endif
-      if (can_endorse_primary() && ticking && r.term == state->current_view)
+      if (!is_retired_committed() && ticking && r.term == state->current_view)
```

Comment:
Reject propose vote requests when we know all future primaries know we are retired.